service_configuration_lib/yaml_cached_view.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from collections import defaultdict
+
+import yaml
+try:
+    from yaml import CSafeLoader as SafeLoader  # type: ignore
+except ImportError:  # pragma: no cover
+    from yaml import SafeLoader  # type: ignore
+
+from service_configuration_lib.cached_view import BaseCachedView
+
+log = logging.getLogger(__name__)
+
+
+class YamlConfigsCachedView(BaseCachedView):
+    """Implementation of yaml configs view.
+    The class keeps dictionary with updated and parsed content of yaml configs.
+    To access files you can use `.configs['service_name']['config_name']`
+
+    Example of usage:
+
+        from service_configuration_lib.cached_view import ConfigsFileWatcher
+        from service_configuration_lib.yaml_cached_view import YamlConfigsCachedView
+
+        watcher = ConfigsFileWatcher(
+        configs_view=YamlConfigsCachedView(),
+            configs_names = ['smartstack', 'authorization'],
+            configs_suffixes=['.yaml']
+        )
+        watcher.process_events()
+        print(watcher.configs_view.configs['schematizer']['smartstack']
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.configs = defaultdict(dict)
+
+    def add(self, path: str, service_name: str, config_name: str, config_suffix: str) -> None:
+        try:
+            with open(path, encoding='utf-8') as fd:
+                self.configs[service_name][config_name] = yaml.load(fd, Loader=SafeLoader)
+        except OSError as exn:
```

Comment:
Can we do something better than just printing warnings?
Because, let's be honest, nobody reads logs unless something horrible just happened.

May be immediately pass these exceptions to the caller of this code to deal with errors?
Or cache exceptions and re-raise then if somebody will try to access the corresponding cached entry?

Reply:
This part is more like user-defined piece to react on filesystem changes. One of examples how to process maintain cache. Hence, it's up to user to write their own implementation instead of this one.
This particular implementation was written to behave in the same way as our current internal code.
I think that's better to leave it for now, once we will figure out another way of caching files and how to deal with exceptions we can add another it.